cloudinit/distros/rhel.py

```diff
@@ -29,6 +29,12 @@ class Distro(distros.Distro):
     network_script_tpl = "/etc/sysconfig/network-scripts/ifcfg-%s"
     tz_local_fn = "/etc/localtime"
     usr_lib_exec = "/usr/libexec"
+    # RHEL and derivatives use NetworkManager DHCP client by default.
```

Comment:
Implemented:
- if the distro-provided dir and regex yield nothing, then try to use the fallback dir and regex. Regex is specifically made to be quite loose (search for any file ending with `.lease` or `.leases`
- Tests for that behavior

Reply:
@phsm Just looking at this change now, I realize that you made `/var/lib/dhclient/` default for all distros.

From what I can tell, the `dhclient` subdirectory is a feature of RHEL derivatives and Arch. Sorry I missed this before.

The following distros use `/var/lib/dhcp/`: including Alpine, Gentoo (and derivatives I assume), Suse (and derivatives), Debian, etc.

No clue what Photon or Mariner use.

With the change as it stands, `/var/lib/dhclient` will be the only directory checked for distros that do not inherit from FreeBSD / RHEL / Debian, so I think that we need to account for those mentioned above.

@igalic is `/var/lib/dhclient` is a reasonable directory for the non-FreeBSD BSDs?

Reply:
I'll need to check.

Reply:
OpenBSD puts it in the same place as FreeBSD:
https://man.openbsd.org/dhclient.8#FILES

NetBSD had switched to `dhcpcd` since 9.0. before that, they put it in the same place, https://man.netbsd.org/NetBSD-8.2/dhclient.8#FILES but with a different name